src/Operators/remapping.jl

```diff
@@ -138,110 +179,82 @@ function overlap_weights(
     return J_ov
 end
 
-function overlap(
-    target::T,
-    source::S,
-) where {
-    T <: SpectralElementSpace1D{<:IntervalTopology},
-    S <: SpectralElementSpace1D{<:IntervalTopology},
-}
-    QS_t = Spaces.quadrature_style(target)
-    QS_s = Spaces.quadrature_style(source)
-    Nq_t = Quadratures.degrees_of_freedom(QS_t)
-    Nq_s = Quadratures.degrees_of_freedom(QS_s)
-    if Nq_t >= Nq_s
-        return overlap_weights(target, source)
-    else
-        J_ov = overlap_weights(source, target)
-        return J_ov'
-    end
-end
-
-function overlap(
-    target::T,
-    source::S,
-) where {
-    T <: SpectralElementSpace2D{<:Topology2D, Quadratures.GL{1}},
-    S <: SpectralElementSpace2D{<:Topology2D, Quadratures.GL{1}},
-}
-    # Calculate element overlap pattern in x-dimension
-    # X_ov[i,j] = overlap length along x-dimension between target elem i and source elem j
-    X_ov = fv_x_overlap(target, source)
-
-    # Calculate element overlap pattern in y-dimension
-    Y_ov = fv_y_overlap(target, source)
-
-    return kron(Y_ov, X_ov)
-end
```

Comment:
Is there any reason to retain special cases for remapping with FV spaces?

Reply:
Wouldn't we want to retain that for the coupling with the ocean or am I not recalling correctly what was the plan?

Reply:
Right; a question is, would we (and can we) define the ClimaCore space that corresponds to the Oceananigans one and feed that into the ClimaCore remapping functionality? As it stands, of course we need to have some adapter to talk to them anyways, so the question is where does that connect to this code that actually computes the remap operator.